content/chainguard/chainguard-enforce/chainguard-enforce-kubernetes/maximum-image-age-policy.md

```diff
@@ -0,0 +1,78 @@
+
+Written in rego (to take advantage of rego’s time functions), this policy imposes a maximum age of 30 days on all images and can be customized for other age limits. The age is measured using the container image's configuration, which has a "created" field. Note that some build tools may fail this check because they build reproducibly, and use a fixed date (e.g. the Unix epoch) as their creation time, but many of these tools support specifying SOURCE_DATE_EPOCH, which aligns the creation time with the date of the source commit.
+
+Here is the policy in full:
+
+```
+apiVersion: policy.sigstore.dev/v1beta1
+kind: ClusterImagePolicy
+metadata:
+  name: maximum-image-age-rego
+  annotations:
+    catalog.chainguard.dev/title: Maximum image age
+    catalog.chainguard.dev/description: |
+      This checks that the maximum age an image is allowed to
+      have is 30 days old.  This is measured using the container
+      image's configuration, which has a "created" field.
+
+      Some build tools may fail this check because they build
+      reproducibly, and use a fixed date (e.g. the Unix epoch)
+      as their creation time, but many of these tools support
+      specifying SOURCE_DATE_EPOCH, which aligns the creation
+      time with the date of the source commit.
+
+    catalog.chainguard.dev/labels: rego
+spec:
+  images: [{ glob: "**" }]
+  authorities: [{ static: { action: pass } }]
+  mode: warn
+  policy:
+    fetchConfigFile: true
+    type: "rego"
+    data: |
+      package sigstore
+
+      nanosecs_per_second = 1000 * 1000 * 1000
+      nanosecs_per_day = 24 * 60 * 60 * nanosecs_per_second
+
+      # Change this to the maximum number of days to allow.
+      maximum_age = 30 * nanosecs_per_day
+
+      isCompliant[response] {
+        created := time.parse_rfc3339_ns(input.config[_].created)
+
+        response := {
+          "result" : time.now_ns() < created + maximum_age,
+          "error" : "Image exceeds maximum allowed age."
+        }
+      }
+```
+
+You can use this policy freely with the open source [Sigstore policy-controller](https://edu.chainguard.dev/open-source/sigstore/policy-controller/how-to-install-policy-controller) to block new deployments of images based on their age.
+
+However, the Maximum Image Age Policy is particularly useful for monitoring the age of artifacts that are _already admitted_ in your workloads. If you are looking for a way to continuously enforce the Maximum Image Age Policy on _all_ existing images in your Kubernetes workloads, you can use Chainguard Enforce. With Chainguard Enforce, all of your images will be continuously evaluated against this policy, even if they passed the policy when initially admitted. You can set Enforce to send a notification (e.g. post to slack, open a github issue) if and when a deployed image falls out of compliance so that corrective action may be taken.
```

Comment:
```suggestion
However, the Maximum Image Age Policy is particularly useful for monitoring the age of artifacts that are _already admitted_ in your workloads. If you are searching for a way to continuously enforce the Maximum Image Age Policy on _all_ existing images in your Kubernetes workloads, you can use Chainguard Enforce. With Chainguard Enforce, all of your images will be continuously evaluated against this policy, even if they passed the policy when initially admitted. You can set Enforce to send a notification (including posting to Slack, or opening a GitHub issue) if and when a deployed image falls out of compliance so that corrective action may be taken.
```